relayer/client-tx.go

```diff
@@ -6,72 +6,93 @@ import (
-	var (
-		clients = &RelayMsgs{Src: []sdk.Msg{}, Dst: []sdk.Msg{}}
-	)
-
 	srcH, dstH, err := UpdatesWithHeaders(c, dst)
 	if err != nil {
 		return err
 	}
 
-	// Create client for the destination chain on the source chain if it doesn't exist
-	if srcCs, err := c.QueryClientState(srcH.Header.Height); err != nil && srcCs == nil {
+	// Create client for the destination chain on the source chain if client id is unspecified
+	if c.PathEnd.ClientID == "" {
 		if c.debug {
 			c.logCreateClient(dst, dstH.Header.Height)
 		}
 		ubdPeriod, err := dst.QueryUnbondingPeriod()
 		if err != nil {
 			return err
 		}
-		consensusParams, err := dst.QueryConsensusParams()
+		msgs := []sdk.Msg{
+			c.PathEnd.CreateClient(
+				dstH,
+				dst.GetTrustingPeriod(),
+				ubdPeriod,
+				c.MustGetAddress(),
+			),
+		}
+		res, success, err := c.SendMsgs(msgs)
 		if err != nil {
 			return err
 		}
-		msg := c.PathEnd.CreateClient(
-			dstH,
-			dst.GetTrustingPeriod(),
-			ubdPeriod,
-			consensusParams,
-			c.MustGetAddress(),
-		)
-		clients.Src = append(clients.Src, msg)
+		if !success {
+			return fmt.Errorf("tx failed: %s", res.RawLog)
+		}
+
+		// update the client identifier
+		// use index 0, the transaction only has one message
+		clientID, err := ParseClientIDFromEvents(res.Logs[0].Events)
+		if err != nil {
+			return err
+		}
+		c.PathEnd.ClientID = clientID
+
+		// TOOD: ensure config file is updated
+
+	} else {
+		// TODO return error if the client does not exist
 	}
 
-	// Create client for the source chain on destination chain if it doesn't exist
-	if dstCs, err := dst.QueryClientState(dstH.Header.Height); err != nil && dstCs == nil {
+	// Create client for the source chain on destination chain if client id is unspecified
+	if dst.PathEnd.ClientID == "" {
 		if dst.debug {
 			dst.logCreateClient(c, srcH.Header.Height)
 		}
 		ubdPeriod, err := c.QueryUnbondingPeriod()
 		if err != nil {
 			return err
 		}
-		consensusParams, err := c.QueryConsensusParams()
+		msgs := []sdk.Msg{
+			dst.PathEnd.CreateClient(
+				srcH,
+				c.GetTrustingPeriod(),
+				ubdPeriod,
+				dst.MustGetAddress(),
+			),
+		}
+		res, success, err := dst.SendMsgs(msgs)
 		if err != nil {
 			return err
 		}
-		msg := dst.PathEnd.CreateClient(
-			srcH,
-			c.GetTrustingPeriod(),
-			ubdPeriod,
-			consensusParams,
-			dst.MustGetAddress(),
-		)
-		clients.Dst = append(clients.Dst, msg)
-	}
+		if !success {
+			return fmt.Errorf("tx failed: %s", res.RawLog)
+		}
 
-	// Send msgs to both chains
-	if clients.Ready() {
-		// TODO: Add retry here for out of gas or other errors
-		if clients.Send(c, dst); clients.Success() {
-			c.Log(fmt.Sprintf("★ Clients created: [%s]client(%s) and [%s]client(%s)",
-				c.ChainID, c.PathEnd.ClientID, dst.ChainID, dst.PathEnd.ClientID))
+		// update client identifier
+		clientID, err := ParseClientIDFromEvents(res.Logs[0].Events)
+		if err != nil {
+			return err
 		}
+		dst.PathEnd.ClientID = clientID
+
+		// TODO: ensure config file is updated
```

Comment:
`if err := overwriteConfig(config); err != nil { return err }` should do this, but `Config` stuff is all in `cmd` package...

Reply:
maybe we make a note on the function that the chains that are passed in have modified `PathEnd`s when they are returned?

Reply:
This was a good suggestion, required minimal changes which was nice